pkg/compression/encoding.go

```diff
@@ -58,7 +56,7 @@ func (e Encoding) String() string {
 	case EncZstd:
 		return "zstd"
 	default:
-		return "unknown"
+		panic(fmt.Sprintf("invalid encoding: %d, supported: %s", e, SupportedEncoding()))
```

Comment:
nit: Panic on a call to String makes me a bit nervous since String can be implicitly called by different standard library functions. Normally in cases like this I would return `fmt.Sprintf("Encoding(%d)", e)` so you can see the value. 

But if you think the risk is low, it's fine to keep it like this 👍

Reply:
I reverted my change